packages/core/src/actors/reducer.ts

```diff
@@ -9,22 +14,33 @@ import { isSCXMLEvent } from '../utils';
  * @returns An actor behavior
  */
 
-export function fromReducer<TState, TEvent extends EventObject>(
+export function fromReducer<TState extends {}, TEvent extends EventObject>(
   transition: (
     state: TState,
     event: TEvent,
     actorContext: ActorContext<TEvent, TState>
   ) => TState,
   initialState: TState
-): ActorBehavior<TEvent, TState> {
-  return {
+): WithRequired<ActorBehavior<TEvent, TState, TState>, 'at'> {
```

Comment:
Do we need this `WithRequired`? It doesn't display that well in the IDe tooltips, cause I see this alias there and it's a lot of noise for the reader.

Fun fact, you can achieve this kind of `WithRequired` through intersections:
```ts
type ActorBehavior = {
  getInitialState: () => void,
  start?: () => void,
  at?: () => void
}

const behavior = {
  getInitialState() {},
} satisfies ActorBehavior & { at: unknown } // error, since `at` has not been provided
```

This doesn't really solve the readability concern. I have some ideas how to solve this, gonna play around with them and post later

Reply:
Yeah, my main goal is to make `at` required, but also, if we make it part of `interpret`, then we're going to remove `at` anyway.

Reply:
Re `WithRequired`:

to display the return type consistently we probably only have 2 options:
```ts
// duplicate
interface ReducerBehavior<TState, TEvent extends EventObject> extends ActorBehavior<TEvent, TState, TState> {
  at: (state: TInternalState) => this;
}

// type gymnastics
type _ReducerBehavior2<TState, TEvent extends EventObject> = ActorBehavior<TEvent, TState, TState> & { at: {} }
interface ReducerBehavior2<TState, TEvent extends EventObject> extends _ReducerBehavior2<TState, TEvent> {}
```

Both preserve JS docs from the original which is good.

> if we make it part of interpret, then we're going to remove at anyway.

Yeah, that's likely true

Reply:
`WithRequired` removed (as well as `at`)